apps/api/v1/views.py

```diff
@@ -50,6 +56,31 @@ def get_queryset(self):
             )
         return qs
 
+    def update(self, request, pk=None):
+        try:
+            obj = self.get_object()
+        except Http404:
+            # Using 406 as the 404 causes middleware issues for the pages.
+            response = Response(status=status.HTTP_406_NOT_ACCEPTABLE)
+        else:
+            if 'like' in request.data:
```

Comment:
Would multiple `@detail_route` methods be better for this?

Reply:
Looks like it is! Let me try to change see if this works as expected.